src/frontend/src/pages/TeamsPage/TeamMembersPageBlock.tsx

```diff
@@ -3,122 +3,168 @@
-  if (setTeamMemberIsError) return <ErrorPage message={setTeamMemberError?.message} />;
-  if (isLoading || setTeamMemberIsLoading || !users) return <LoadingIndicator />;
+  const toast = useToast();
 
-  const handleSubmit = async () => {
-    try {
-      await mutateAsync(members.map((member) => member.id));
-      setIsEditingMembers(false);
-    } catch (error) {
-      alert(error);
-    }
-  };
+  if (allUsersIsError) return <ErrorPage message={allUsersError?.message} />;
+  if (allUsersIsLoading || setTeamMembersIsLoading || setTeamHeadIsLoading || !users) return <LoadingIndicator />;
 
   const hasPerms = auth.user && (isAdmin(auth.user.role) || auth.user.userId === team.head.userId);
 
-  const options = users
+  const memberOptions = users
     .filter((user) => user.userId !== team.head.userId && !team.leads.map((lead) => lead.userId).includes(user.userId))
     .sort((a, b) => (a.firstName > b.firstName ? 1 : -1))
     .map(userToAutocompleteOption);
 
-  const editButtons = (
-    <div style={{ display: 'flex' }}>
-      <NERFailButton
-        onClick={() => {
-          setIsEditingMembers(false);
-          setMembers(team.members.map(userToAutocompleteOption));
-        }}
-      >
-        Cancel
-      </NERFailButton>
-      <NERSuccessButton sx={{ ml: 2 }} onClick={handleSubmit}>
-        Save
-      </NERSuccessButton>
-    </div>
-  );
+  const headOptions = users
+    .filter((user) => memberOptions.some((option) => parseInt(option.id) === user.userId) && isHead(user.role))
+    .sort((a, b) => (a.firstName > b.firstName ? 1 : -1))
+    .map(userToAutocompleteOption);
+
+  const submitHead = async () => {
+    try {
+      await setTeamHeadMutateAsync(parseInt(head.id));
+      setIsEditingHead(false);
+    } catch (error) {
+      error instanceof Error && toast.error(error.message);
+    }
+  };
+
+  const submitMembers = async () => {
+    try {
+      await setTeamMembersMutateAsync(members.map((member) => parseInt(member.id)));
+      setIsEditingMembers(false);
+    } catch (error) {
+      error instanceof Error && toast.error(error.message);
+    }
+  };
 
-  const editingView = (
-    <PageBlock title={'People'} headerRight={editButtons}>
-      <Grid container spacing={1}>
-        <Grid item xs={12}>
-          <DetailDisplay label="Head" content={fullNamePipe(team.head)} />
+  const EditingHeadView = () => (
+    <Grid container direction={'row'} spacing={1}>
+      <Grid item>
+        <Typography sx={{ fontWeight: 'bold' }} display="inline">
+          Head:
+        </Typography>
+      </Grid>
+      <Grid item xs={true} mt={-1}>
+        <NERAutocomplete
+          id="head-autocomplete"
+          options={headOptions}
+          onChange={(_event, newValue) => newValue && setHead(newValue)}
+          filterSelectedOptions
+          size="small"
+          placeholder="Select a User"
+          value={head}
+        />
+      </Grid>
+      <Grid container direction={'row'} xs={4} md={2} lg={1}>
```

Comment:
nitpick: xs can be 3 here, other stuff starts looking jank when you get small enough for it to need to be 4